In [4]:
import pandas as pd
import numpy as np

In [6]:
df = pd.read_csv('empleados.csv', delimiter=";")
df

Nombre  Apellido       Sexo   Estado_Civil  Edad  Experiencia  \
0    Alexandro       NaN  Masculino     Soltero(a)  22.0          3.0   
1       Carlos       NaN  Masculino      Casado(a)  38.0         10.0   
2       Felipa       NaN   Femenino     Soltero(a)  26.0          5.0   
3       Daniel       NaN  Masculino      Casado(a)  35.0         10.0   
4         John       NaN  Masculino      Casado(a)  35.0         13.0   
..         ...       ...        ...            ...   ...          ...   
246       Jose       NaN  Masculino      Casado(a)  30.0          6.0   
247    Marcelo       NaN  Masculino  Divorciado(a)  44.0         15.0   
248       Eric       NaN        NaN      Casado(a)  25.0          7.0   
249       Marc       NaN        NaN      Casado(a)  24.0          2.0   
250    Leandro       NaN  Masculino      Casado(a)  37.0         11.0   

     Salario Gerencia          Departamento  
0     2423.0       No            Publicidad  
1     3132.0       Sí                   NaN  
2     2639.0       No             Comercial  
3     3122.0       No             Comercial  
4     3431.0       Sí  Servicios al cliente  
..       ...      ...                   ...  
246   2760.0       No  Servicios al cliente  
247   3616.0       No                    IT  
248   2880.0       Sí      Recursos Humanos  
249   2172.0       No              Producto  
250   3245.0       No              Finanzas  

[251 rows x 9 columns]

In [7]:
df.dtypes

Nombre           object
Apellido        float64
Sexo             object
Estado_Civil     object
Edad            float64
Experiencia     float64
Salario         float64
Gerencia         object
Departamento     object
dtype: object

In [16]:
df.isnull().sum()

Nombre            6
Apellido        251
Sexo             37
Estado_Civil      4
Edad             61
Experiencia       4
Salario           6
Gerencia         21
Departamento     12
dtype: int64

#primero quitamos la columna appellidos, otra manera seria usar columns = 'Apellido'

In [14]:
df2 = df.drop('Apellido', axis=1)
df2

Nombre       Sexo   Estado_Civil  Edad  Experiencia  Salario Gerencia  \
0    Alexandro  Masculino     Soltero(a)  22.0          3.0   2423.0       No   
1       Carlos  Masculino      Casado(a)  38.0         10.0   3132.0       Sí   
2       Felipa   Femenino     Soltero(a)  26.0          5.0   2639.0       No   
3       Daniel  Masculino      Casado(a)  35.0         10.0   3122.0       No   
4         John  Masculino      Casado(a)  35.0         13.0   3431.0       Sí   
..         ...        ...            ...   ...          ...      ...      ...   
246       Jose  Masculino      Casado(a)  30.0          6.0   2760.0       No   
247    Marcelo  Masculino  Divorciado(a)  44.0         15.0   3616.0       No   
248       Eric        NaN      Casado(a)  25.0          7.0   2880.0       Sí   
249       Marc        NaN      Casado(a)  24.0          2.0   2172.0       No   
250    Leandro  Masculino      Casado(a)  37.0         11.0   3245.0       No   

             Departamento  
0              Publicidad  
1                     NaN  
2               Comercial  
3               Comercial  
4    Servicios al cliente  
..                    ...  
246  Servicios al cliente  
247                    IT  
248      Recursos Humanos  
249              Producto  
250              Finanzas  

[251 rows x 8 columns]

Eliminamos las lineas que tienen un valor vacio

In [15]:
df3 = df2.dropna()
df3

Nombre       Sexo   Estado_Civil  Edad  Experiencia  Salario Gerencia  \
0    Alexandro  Masculino     Soltero(a)  22.0          3.0   2423.0       No   
2       Felipa   Femenino     Soltero(a)  26.0          5.0   2639.0       No   
3       Daniel  Masculino      Casado(a)  35.0         10.0   3122.0       No   
4         John  Masculino      Casado(a)  35.0         13.0   3431.0       Sí   
6      Ruperta   Femenino      Casado(a)  54.0         18.0   3962.0       No   
..         ...        ...            ...   ...          ...      ...      ...   
244   Hernando  Masculino  Divorciado(a)  29.0          4.0   2510.0       No   
245       Vito  Masculino  Divorciado(a)  22.0          4.0   2530.0       No   
246       Jose  Masculino      Casado(a)  30.0          6.0   2760.0       No   
247    Marcelo  Masculino  Divorciado(a)  44.0         15.0   3616.0       No   
250    Leandro  Masculino      Casado(a)  37.0         11.0   3245.0       No   

             Departamento  
0              Publicidad  
2               Comercial  
3               Comercial  
4    Servicios al cliente  
6                Producto  
..                    ...  
244                    IT  
245              Producto  
246  Servicios al cliente  
247                    IT  
250              Finanzas  

[145 rows x 8 columns]

Ahora se Dejaran las lineas que tengan mas de 5 valores validos

In [22]:
df4 = df2.dropna(axis=0, thresh=5)
df4.shape
df4

Nombre       Sexo   Estado_Civil  Edad  Experiencia  Salario Gerencia  \
0    Alexandro  Masculino     Soltero(a)  22.0          3.0   2423.0       No   
1       Carlos  Masculino      Casado(a)  38.0         10.0   3132.0       Sí   
2       Felipa   Femenino     Soltero(a)  26.0          5.0   2639.0       No   
3       Daniel  Masculino      Casado(a)  35.0         10.0   3122.0       No   
4         John  Masculino      Casado(a)  35.0         13.0   3431.0       Sí   
..         ...        ...            ...   ...          ...      ...      ...   
246       Jose  Masculino      Casado(a)  30.0          6.0   2760.0       No   
247    Marcelo  Masculino  Divorciado(a)  44.0         15.0   3616.0       No   
248       Eric        NaN      Casado(a)  25.0          7.0   2880.0       Sí   
249       Marc        NaN      Casado(a)  24.0          2.0   2172.0       No   
250    Leandro  Masculino      Casado(a)  37.0         11.0   3245.0       No   

             Departamento  
0              Publicidad  
1                     NaN  
2               Comercial  
3               Comercial  
4    Servicios al cliente  
..                    ...  
246  Servicios al cliente  
247                    IT  
248      Recursos Humanos  
249              Producto  
250              Finanzas  

[246 rows x 8 columns]

De esta manera encontramos que tenemos 245  lineas con datos con mas de 5 columnas con valores y 145 completos del total de 251, ahora faltaria revisar si no hay valores duplicados y eliminarlos del registro bueno, que en este caso es el df3

In [29]:
print(df3.shape)
duplicados = df3.duplicated()
print(df3[duplicados]) # de esta manera sale en tabla

df5 = df3.drop_duplicates()
df5


(145, 8)
    Nombre       Sexo Estado_Civil  Edad  Experiencia  Salario Gerencia  \
14  Selena   Femenino    Casado(a)  24.0          6.0   2709.0       Sí   
65   Lucio  Masculino   Soltero(a)  54.0         12.0   3317.0       No   

        Departamento  
14      Contabilidad  
65  Recursos Humanos  


Nombre       Sexo   Estado_Civil  Edad  Experiencia  Salario Gerencia  \
0    Alexandro  Masculino     Soltero(a)  22.0          3.0   2423.0       No   
2       Felipa   Femenino     Soltero(a)  26.0          5.0   2639.0       No   
3       Daniel  Masculino      Casado(a)  35.0         10.0   3122.0       No   
4         John  Masculino      Casado(a)  35.0         13.0   3431.0       Sí   
6      Ruperta   Femenino      Casado(a)  54.0         18.0   3962.0       No   
..         ...        ...            ...   ...          ...      ...      ...   
244   Hernando  Masculino  Divorciado(a)  29.0          4.0   2510.0       No   
245       Vito  Masculino  Divorciado(a)  22.0          4.0   2530.0       No   
246       Jose  Masculino      Casado(a)  30.0          6.0   2760.0       No   
247    Marcelo  Masculino  Divorciado(a)  44.0         15.0   3616.0       No   
250    Leandro  Masculino      Casado(a)  37.0         11.0   3245.0       No   

             Departamento  
0              Publicidad  
2               Comercial  
3               Comercial  
4    Servicios al cliente  
6                Producto  
..                    ...  
244                    IT  
245              Producto  
246  Servicios al cliente  
247                    IT  
250              Finanzas  

[143 rows x 8 columns]

El df5 es el dataframe que no tiene valores duplicados

En ocasiones salen valores NAN or NONe si es que se decidio omitir la razon de algun valor por alguna razon por lo que en ocasiones es necesario rellenar o predecir los valores. Para el metodo de imputacion se utilizo otro df que solo tenia valores faltantes en la edad, aqui se improvisara con el df4

In [31]:
df4['Edad'] = df4['Edad'].fillna(df4['Edad'].mean())
print(df4.isnull().sum())

Nombre           4
Sexo            33
Estado_Civil     0
Edad             0
Experiencia      1
Salario          2
Gerencia        16
Departamento     8
dtype: int64


C:\Users\Daniel\AppData\Local\Temp\ipykernel_18212\3443425672.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df4['Edad'] = df4['Edad'].fillna(df4['Edad'].mean())


El otro metodo a trabajar el de imputacion de regresion lineal, pero ese usa un metodo mucho mas distinto de dataframe, por lo que tendras que cambiar de branch al 01_07i para ver la solucion